docs/ops/upgrading.md

```diff
@@ -227,15 +229,25 @@ Savepoints are compatible across Flink versions as indicated by the table below:
           <td class="text-center"></td>
           <td class="text-center">O</td>
           <td class="text-center">O</td>
-          <td class="text-left">When migrating from Flink 1.2.x to Flink 1.3.x, changing parallelism at the same
-          time is not supported. Users have to first take a savepoint after migrating to Flink 1.3.x, and then change
-          parallelism.</td>
+          <td class="text-center">O</td>
+          <td class="text-left">When migrating from Flink 1.2.x to Flink 1.3.x+, changing parallelism at the same
+          time is not supported. Users have to first take a savepoint after migrating to Flink 1.3.x+, and then change
+          parallelism. Savepoints taken by the CEP library cannot be restored in 1.4.x+.</td>
```

Comment:
"taken by" sounds a bit strange, because it is not the CEP library that takes them.

Reply:
How about `Savepoints taken for CEP applications ...`?

Reply:
Another suggestion : `Savepoints taken for applications using CEP library.`